src/sql/workbench/api/common/extHostModelViewDialog.ts

```diff
@@ -600,25 +617,27 @@ class WizardImpl implements azdata.window.Wizard {
 	}
 }
 
-class ModelViewDashboardImpl implements azdata.window.ModelViewDashboard {
+class ModelViewDashboardImpl extends Disposable implements azdata.window.ModelViewDashboard {
 	private _tabbedPanel: azdata.TabbedPanelComponent;
 	private _view: azdata.ModelView;
 
 	constructor(
 		private _editor: ModelViewEditorImpl,
 		private _options?: azdata.ModelViewDashboardOptions
 	) {
+		super();
+		this._register(this._editor);
 	}
 
-	updateTabs(tabs: (azdata.DashboardTab | azdata.DashboardTabGroup)[]): void {
+	updateTabs(tabs: (DisposableDashboardTab | DisposableDashboardTabGroup)[]): void {
```

Comment:
These public APIs (updateTabs and registerTabs) shouldn't need to change since they're only called by the extensions and will only be expected to pass in the azdata.Tab/TabGroup objects.

Reply:
This is on the implementation side, exposed APIs are unchanged in azdata.

Reply:
These should never be called internally (or if they are we should make a separate internal-only implementation of it or something). And by using this typing you're creating a scenario where we think that the types are disposable - but whenever called by an extension they won't actually have dispose implemented so can/will cause problems later on.

Reply:
I hope I fixed it right ?